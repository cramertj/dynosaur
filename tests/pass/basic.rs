use dynosaur;

#[dynosaur::make_dyn(dyn)]
trait MyTrait {
    type Item;
    async fn foo(&self) -> Self::Item;
}

fn main() {}
