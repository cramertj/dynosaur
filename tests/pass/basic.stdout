#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use dynosaur;

trait MyTrait {
    type Item;
    async fn foo(&self)
    -> Self::Item;
}
trait ErasedMyTrait {
    type Item;
    fn foo(&self)
    -> Self::Item;
}

fn main() {}
